scripts/regenerate_protos.sh

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+
+set -e -o pipefail
+
+echo "Regenerating Protocol Buffers"
+
```

Comment:
so we've talked about this a little and i think i understand the issue 

i think the comments below are super helpful wrt having there be two different outputs - it would be sorta nice if the motivation was at the top of this script as well, which i think is described pretty reasonably in the description of this PR

Reply:
This is actually because we need to reference these relative to where they are. 

JS OUT DIR is where the compiled typescript goes, and the TS OUT DIR is where the typescript code running with ts-node will find the files.